### DISCUSSION TITLE
Request: amount of GPU memory

### DISCUSSION THREAD

**Discussion by JonnyTech** [turkaysoft/glow#27] [Ideas]
Is it possible to display the amount of memory per graphics card(s) in the report, please?

**Comment by turkaysoft** [comment]
Hello,

Pulling the amount of Graphics Card VRAM is complicated. Because Glow does not include any libraries.

I did a lot of tests to write this in Vanilla C# but I still have problems with multi-GPU.

So unfortunately it is not possible in the near future.

**Comment by JonnyTech** [comment]
Thanks for the explaination.

I found a post which claims that the value is stored in the Windows registry. Is it possible to include that?

https://superuser.com/questions/1461858/fetch-correct-vram-for-gpu-via-command-line-on-windows

https://github.com/farag2/PC-information/blob/master/PC.ps1#L174

**Comment by JonnyTech** [comment]
It does indeed show the correct amount of RAM in the registry key, whereas `msinfo32.exe` is incorrect:

![image](https://github.com/roines45/glow/assets/3584732/fbcad865-3d88-40d1-ac52-4dc3cfc47368)

The computer has two graphics cards, 0000 is inbuilt to the motherboard and does not have any GPU RAM and 0001 is a NVidia card installed into a PCIe slot.

Hope that helps.

**Comment by turkaysoft** [comment]
Testing is in progress.
Unfortunately, I can't give you the exact date, but I want you to know that I want to add it.

**Reply by turkaysoft** [reply]
Hello again! @JonnyTech 

I managed to integrate it into the interface correctly and quickly, and the Levenshtein Distance algorithm was very helpful in this regard, of course.

What's more, it supports multiple GPUs, whether AMD, Intel, or NVIDIA. It can detect any type of internal or external device. And it does so without the need for external libraries.

In the next patch, the amount of VRAM will be added to the interface.

<img width="797" height="234" alt="image" src="https://github.com/user-attachments/assets/5d202be9-81e5-4cac-9427-900b44e9d4b6" />

<img width="796" height="243" alt="image" src="https://github.com/user-attachments/assets/87c127ee-e121-4da3-b345-63364a83ce10" />

**Comment by JonnyTech** [comment]
Great work, thank you @turkaysoftware!
